dev/tasks/crossbow.py

```diff
@@ -301,6 +343,56 @@ def from_repo(cls, repo):
                    version=version)
 
 
+def _filename_from_url(url):
+    parsed = urllib.parse.urlparse(url)
+    return Path(parsed.path).name
+
+
+class DistTarget(Target):
+    """Describes target source archive the builds run against
+
+    This serializable data container holding information about arrow's
+    source distribution, tipically an apache release.
+    """
+
+    def __init__(self, archive_url, verify=True, **kwargs):
+        assert archive_url.endswith('.tar.gz')
+        self.archive_url = archive_url
+        self.verify = verify
+        super().__init__(**kwargs)
+
+    @property
+    def source(self):
+        return 'dist'
+
+    @property
+    def signature_url(self):
+        return self.archive_url + '.asc'
+
+    @property
+    def checksum_url(self):
+        """SHA512 checksum"""
+        return self.archive_url + '.sha512'
```

Comment:
There are going to be multiple checksums here: sha1, sha256, and sha512 so we should handle that.

Reply:
Can you change this to `.sha256`? As of ARROW-2934 we will only be generating `.sha256` and not `.sha512`.

Reply:
sure